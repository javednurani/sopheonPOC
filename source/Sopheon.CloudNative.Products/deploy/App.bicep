param sites_StratusProductManagement_Dev_name string = 'StratusProductManagement-Dev'
param serverfarms_StratusProductManagement_AppPlan_Dev_externalid string = '/subscriptions/1c4bef1d-8a40-4a6d-96d6-764bb466ac46/resourceGroups/Accolade-Next-Gen/providers/Microsoft.Web/serverfarms/StratusProductManagement-AppPlan-Dev'

resource sites_StratusProductManagement_Dev_name_resource 'Microsoft.Web/sites@2021-02-01' = {
  name: sites_StratusProductManagement_Dev_name
  location: 'West US'
  tags: {
    'Environment Type': 'Research'
    Owner: 'CloudTeam-1'
    'Review Date': '06-30-21'
  }
  kind: 'app'
  properties: {
    enabled: true
    hostNameSslStates: [
      {
        name: 'stratusproductmanagement-dev.azurewebsites.net'
        sslState: 'Disabled'
        hostType: 'Standard'
      }
      {
        name: 'stratusproductmanagement-dev.scm.azurewebsites.net'
        sslState: 'Disabled'
        hostType: 'Repository'
      }
    ]
    serverFarmId: serverfarms_StratusProductManagement_AppPlan_Dev_externalid
    reserved: false
    isXenon: false
    hyperV: false
    siteConfig: {
      numberOfWorkers: 1
      acrUseManagedIdentityCreds: false
      alwaysOn: true
      http20Enabled: false
      functionAppScaleLimit: 0
      minimumElasticInstanceCount: 1
      appSettings: {
        
      }
    }
    scmSiteAlsoStopped: false
    clientAffinityEnabled: true
    clientCertEnabled: false
    clientCertMode: 'Required'
    hostNamesDisabled: false
    customDomainVerificationId: 'F50C4A4633DCAF18B0C09B554884823C47C074E58674E017191D38260A145C6A'
    containerSize: 0
    dailyMemoryTimeQuota: 0
    httpsOnly: false
    redundancyMode: 'None'
    storageAccountRequired: false
    keyVaultReferenceIdentity: 'SystemAssigned'
  }
}

resource sites_StratusProductManagement_Dev_name_ftp 'Microsoft.Web/sites/basicPublishingCredentialsPolicies@2021-02-01' = {
  parent: sites_StratusProductManagement_Dev_name_resource
  name: 'ftp'
  properties: {
    allow: true
  }
}

resource sites_StratusProductManagement_Dev_name_scm 'Microsoft.Web/sites/basicPublishingCredentialsPolicies@2021-02-01' = {
  parent: sites_StratusProductManagement_Dev_name_resource
  name: 'scm'
  properties: {
    allow: true
  }
}

resource sites_StratusProductManagement_Dev_name_web 'Microsoft.Web/sites/config@2021-02-01' = {
  parent: sites_StratusProductManagement_Dev_name_resource
  name: 'web'
  properties: {
    numberOfWorkers: 1
    defaultDocuments: [
      'Default.htm'
      'Default.html'
      'Default.asp'
      'index.htm'
      'index.html'
      'iisstart.htm'
      'default.aspx'
      'index.php'
      'hostingstart.html'
    ]
    netFrameworkVersion: 'v5.0'
    requestTracingEnabled: true
    requestTracingExpirationTime: '12/31/9999 11:59:00 PM'
    remoteDebuggingEnabled: false
    remoteDebuggingVersion: 'VS2019'
    httpLoggingEnabled: true
    acrUseManagedIdentityCreds: false
    logsDirectorySizeLimit: 35
    detailedErrorLoggingEnabled: true
    publishingUsername: '$StratusProductManagement-Dev'
    scmType: 'None'
    use32BitWorkerProcess: true
    webSocketsEnabled: false
    alwaysOn: true
    managedPipelineMode: 'Integrated'
    virtualApplications: [
      {
        virtualPath: '/'
        physicalPath: 'site\\wwwroot'
        preloadEnabled: true
      }
    ]
    loadBalancing: 'LeastRequests'
    experiments: {
      rampUpRules: []
    }
    autoHealEnabled: false
    vnetRouteAllEnabled: false
    vnetPrivatePortsCount: 0
    localMySqlEnabled: false
    ipSecurityRestrictions: [
      {
        ipAddress: 'Any'
        action: 'Allow'
        priority: 1
        name: 'Allow all'
        description: 'Allow all access'
      }
    ]
    scmIpSecurityRestrictions: [
      {
        ipAddress: 'Any'
        action: 'Allow'
        priority: 1
        name: 'Allow all'
        description: 'Allow all access'
      }
    ]
    scmIpSecurityRestrictionsUseMain: false
    http20Enabled: false
    minTlsVersion: '1.2'
    scmMinTlsVersion: '1.0'
    ftpsState: 'AllAllowed'
    preWarmedInstanceCount: 0
    functionAppScaleLimit: 0
    healthCheckPath: '/swagger/ui'
    functionsRuntimeScaleMonitoringEnabled: false
    minimumElasticInstanceCount: 1
    azureStorageAccounts: {}
  }
}

resource sites_StratusProductManagement_Dev_name_sites_StratusProductManagement_Dev_name_azurewebsites_net 'Microsoft.Web/sites/hostNameBindings@2021-02-01' = {
  parent: sites_StratusProductManagement_Dev_name_resource
  name: '${sites_StratusProductManagement_Dev_name}.azurewebsites.net'
  properties: {
    siteName: 'StratusProductManagement-Dev' //TODO: Configure for TokenReplace
    hostNameType: 'Verified'
  }
}

resource sites_StratusProductManagement_Dev_name_Microsoft_AspNetCore_AzureAppServices_SiteExtension 'Microsoft.Web/sites/siteextensions@2021-02-01' = {
  parent: sites_StratusProductManagement_Dev_name_resource
  name: 'Microsoft.AspNetCore.AzureAppServices.SiteExtension'
}
