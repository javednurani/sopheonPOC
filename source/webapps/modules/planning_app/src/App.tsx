import React from 'react';

import { AppDispatchProps, AppStateProps } from './AppContainer';
import { AppProps } from './TempSopheonShellAPI';

export type Props = AppProps<AppStateProps, AppDispatchProps>;

const App: React.FunctionComponent<Props> = () => <div></div>;

export default App;
