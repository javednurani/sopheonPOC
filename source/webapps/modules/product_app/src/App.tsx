import { registerIcons, Stack } from '@fluentui/react';
import { AppProps } from '@sopheon/shell-api';
import React, { useEffect } from 'react';

import { AppDispatchProps, AppStateProps } from './AppContainer';
import { ReactComponent as AeroIndustry } from './images/industryico_Aero.svg';
import { ReactComponent as AgIndustry } from './images/industryico_Ag.svg';
import { ReactComponent as AutoIndustry } from './images/industryico_Auto.svg';
import { ReactComponent as ConstREIndustry } from './images/industryico_ConstRE.svg';
import { ReactComponent as ConsumerIndustry } from './images/industryico_Consumer.svg';
import { ReactComponent as EdIndustry } from './images/industryico_Ed.svg';
import { ReactComponent as EnergyIndustry } from './images/industryico_Energy.svg';
import { ReactComponent as FinIndustry } from './images/industryico_Fin.svg';
import { ReactComponent as GovtIndustry } from './images/industryico_Govt.svg';
import { ReactComponent as HealthIndustry } from './images/industryico_Health.svg';
import { ReactComponent as HospIndustry } from './images/industryico_Hosp.svg';
import { ReactComponent as IndusIndustry } from './images/industryico_Indus.svg';
import { ReactComponent as MediaIndustry } from './images/industryico_Media.svg';
import { ReactComponent as MemberIndustry } from './images/industryico_Member.svg';
import { ReactComponent as ServicesIndustry } from './images/industryico_Services.svg';
import { ReactComponent as TechIndustry } from './images/industryico_Tech.svg';
import { ReactComponent as TeleIndustry } from './images/industryico_Tele.svg';
import { ReactComponent as TransIndustry } from './images/industryico_Trans.svg';
import OnboardingInfo from './onboardingInfo';
import { EnvironmentScopedApiRequestDto } from './types';

export type Props = AppProps<AppStateProps, AppDispatchProps>;

const svgIconStyle: React.CSSProperties = {
  width: '20px',
  height: '20px',
  overflow: 'visible',
};
registerIcons({
  icons: {
    AgIndustryIcon: <AgIndustry style={svgIconStyle} />,
    AeroIndustryIcon: <AeroIndustry style={svgIconStyle} />,
    AutoIndustryIcon: <AutoIndustry style={svgIconStyle} />,
    ConstREIndustryIcon: <ConstREIndustry style={svgIconStyle} />,
    ConsumerIndustryIcon: <ConsumerIndustry style={svgIconStyle} />,
    EduIndustryIcon: <EdIndustry style={svgIconStyle} />,
    EnergyIndustryIcon: <EnergyIndustry style={svgIconStyle} />,
    FinIndustryIcon: <FinIndustry style={svgIconStyle} />,
    GovtIndustryIcon: <GovtIndustry style={svgIconStyle} />,
    HealthIndustryIcon: <HealthIndustry style={svgIconStyle} />,
    HospIndustryIcon: <HospIndustry style={svgIconStyle} />,
    IndusIndustryIcon: <IndusIndustry style={svgIconStyle} />,
    MediaIndustryIcon: <MediaIndustry style={svgIconStyle} />,
    MemberIndustryIcon: <MemberIndustry style={svgIconStyle} />,
    ServicesIndustryIcon: <ServicesIndustry style={svgIconStyle} />,
    TechIndustryIcon: <TechIndustry style={svgIconStyle} />,
    TeleIndustryIcon: <TeleIndustry style={svgIconStyle} />,
    TransIndustryIcon: <TransIndustry style={svgIconStyle} />,
  },
});

const App: React.FunctionComponent<Props> = ({
  currentStep,
  createProduct,
  updateProduct,
  getProducts,
  environmentKey,
  getAccessToken,
  accessToken,
}: Props) => {
  useEffect(() => {
    // getAccessToken triggers Shell action to store access token, freshly acquired from MSAL, in Redux state
    // after getAccessToken is called (here, on ProductApp render), shellApi::accessToken should be up-to-date access token from MSAL.acquireTokenSilent()
    getAccessToken();

    // get any Products for User on initial load of Product App (will determine Onboarding State, ____, etc)
    const requestDto: EnvironmentScopedApiRequestDto = {
      EnvironmentKey: environmentKey,
      AccessToken: accessToken,
    };

    getProducts(requestDto);
  }, []);

  return (
    <div>
      <Stack>
        <OnboardingInfo
          currentStep={currentStep}
          createProduct={createProduct}
          updateProduct={updateProduct}
          environmentKey={environmentKey}
          accessToken={accessToken}
        />
      </Stack>
    </div>
  );
};
export default App;
