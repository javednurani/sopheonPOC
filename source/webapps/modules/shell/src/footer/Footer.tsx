import { Icon, IIconStyles, Stack, Sticky, StickyPositionType } from '@fluentui/react';
import React from 'react';
import { FormattedMessage } from 'react-intl';

export interface IFooterProps {
  showFooter: boolean;
}

const Footer: React.FunctionComponent<IFooterProps> = ({ showFooter }: IFooterProps) => {
  const footerStyle: React.CSSProperties = {
    borderTop: '1px solid',
    padding: '10px',
  };

  const iconStyles: Partial<IIconStyles> = {
    root: {
      marginRight: '8px',
      marginLeft: '8px',
    },
  };

  if (showFooter) {
    return (
      <Sticky stickyPosition={StickyPositionType.Footer}>
        <footer style={footerStyle} role="contentinfo">
          <Stack horizontal verticalAlign="center">
            <Stack.Item grow align="start">
              <FormattedMessage id="footer.copyright" />
            </Stack.Item>
            <Stack.Item shrink align="end">
              <span>
                <Icon styles={iconStyles} iconName={'LikeSolid'} aria-hidden="true" />
                <Icon styles={iconStyles} iconName={'CommentSolid'} aria-hidden="true" />
                <Icon styles={iconStyles} iconName={'LinkedInLogo'} aria-hidden="true" />
                <Icon styles={iconStyles} iconName={'TeamsLogo'} aria-hidden="true" />
              </span>
            </Stack.Item>
          </Stack>
        </footer>
      </Sticky>
    );
  }
  return <></>;
};

export default Footer;
