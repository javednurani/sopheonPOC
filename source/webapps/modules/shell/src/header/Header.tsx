import { IStackTokens, Stack, Sticky, StickyPositionType } from '@fluentui/react';
import { GetAccessTokenAction } from '@sopheon/shell-api';
import React, { FunctionComponent } from 'react';
import { useIntl } from 'react-intl';

import { SetEnvironmentKeyAction } from '../authentication/authReducer';
import LoginSignupButton from '../authentication/LoginSignupButton';
import Navbar from '../navbar/Navbar';
import SopheonLogo from '../SopheonLogo';
import ThemeSelector from '../themes/components/themeSelector/ThemeSelector';
import { ChangeThemeAction } from '../themes/themeReducer/themeReducer';

interface HeaderProps {
  changeTheme: (useDarkTheme: boolean) => ChangeThemeAction;
  setEnvironmentKey: (environmentKey: string) => SetEnvironmentKeyAction;
  getAccessToken: () => GetAccessTokenAction;
}

const Header: FunctionComponent<HeaderProps> = ({ changeTheme, setEnvironmentKey, getAccessToken }: HeaderProps) => {
  const { formatMessage } = useIntl();

  const headerStyle: React.CSSProperties = {
    marginTop: '8px',
    borderBottom: '1px solid',
  };

  const logoStyle: React.CSSProperties = {
    height: '100%',
    width: '100%',
    maxWidth: '200px',
    maxHeight: '200px',
    minWidth: '25px',
    minHeight: '25px',
    overflow: 'visible',
  };

  const logoContainerStyle: React.CSSProperties = {
    margin: '5px',
  };

  const stackTokensWithGap: IStackTokens = {
    childrenGap: 2,
  };

  /*
  const location = useLocation();

  const getTitle = (path: string): string => {
    const currentApp: AppModule | undefined = appModules.find(appModule => appModule.routeName === path);
    return currentApp ? formatMessage({ id: currentApp.displayNameResourceKey }) : formatMessage({ id: 'defaultTitle' });
  };

  const pageTitle: string = getTitle(location.pathname);
  */

  return (
    <Sticky stickyPosition={StickyPositionType.Header} isScrollSynced>
      <header style={headerStyle} role="banner">
        <Stack horizontal verticalAlign="center">
          <Stack.Item shrink>
            <div style={logoContainerStyle} title={formatMessage({ id: 'sopheon' })}>
              <SopheonLogo style={logoStyle} />
            </div>
          </Stack.Item>
          <Stack.Item>
            <Navbar />
          </Stack.Item>
          <Stack.Item grow align="center">
            <h1 id="page-title" className="page-title">
              {/* {pageTitle} */}
            </h1>
          </Stack.Item>
          <Stack.Item>
            <Stack tokens={stackTokensWithGap}>
              <Stack.Item>
                <LoginSignupButton setEnvironmentKey={setEnvironmentKey} getAccessToken={getAccessToken} />
              </Stack.Item>
              <Stack.Item>
                <ThemeSelector changeTheme={changeTheme} />
              </Stack.Item>
            </Stack>
          </Stack.Item>
        </Stack>
      </header>
    </Sticky>
  );
};

export default Header;
