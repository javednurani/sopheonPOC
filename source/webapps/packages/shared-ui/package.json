{
  "name": "@sopheon/shared-ui",
  "version": "1.0.0",
  "description": "A package of UI Components for Sopheon",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "tsc",
    "test": ""
  },
  "keywords": [],
  "author": "Sopheon",
  "license": "ISC",
  "dependencies": {
    "@fluentui/react": "^7.155.4",
    "@fluentui/react-theme-provider": "^0.18.5",
    "@microsoft/applicationinsights-react-js": "^3.1.0",
    "@microsoft/applicationinsights-web": "^2.6.1",
    "react": "^16.14.0",
    "react-dom": "^16.14.0",
    "react-intl": "^5.13.4",
    "typescript": "^4.2.2"
  },
  "devDependencies": {
    "@types/node": "^15.0.1",
    "@typescript-eslint/eslint-plugin": "^4.14.0",
    "@typescript-eslint/parser": "^4.14.0",
    "babel-eslint": "^10.1.0",
    "core-js": "^3.9.1",
    "eslint": "^7.14.0",
    "eslint-config-prettier": "6.11.0",
    "eslint-config-sopheon": "^1.0.7",
    "eslint-plugin-html": "^6.1.2",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-prettier": "^3.3.1",
    "eslint-plugin-react": "^7.20.0",
    "eslint-plugin-react-hooks": "^4.0.4",
    "eslint-plugin-simple-import-sort": "^7.0.0",
    "prettier": "^2.0.5",
    "prettier-eslint": "^11.0.0",
    "stylelint": "^13.6.1",
    "stylelint-config-prettier": "^8.0.2",
    "stylelint-config-sass-guidelines": "^7.0.0",
    "stylelint-order": "^4.1.0",
    "stylelint-prettier": "^1.1.2"
  }
}
