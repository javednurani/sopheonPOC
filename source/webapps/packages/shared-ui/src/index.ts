export * from './errorBoundary/ErrorBoundary';
export * from './language/constants';
export * from './language/languages';
export * from './language/messages';
export * from './language/types';
export * from './theme/themes';
