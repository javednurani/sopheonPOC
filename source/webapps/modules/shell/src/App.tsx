import { initializeIcons, registerIcons, ScrollablePane, ScrollbarVisibility, Stack } from '@fluentui/react';
import { useTheme } from '@fluentui/react-theme-provider';
import React, { CSSProperties, FunctionComponent } from 'react';
import { useIntl } from 'react-intl';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import IdleMonitor from './authentication/IdleMonitor';
import Login from './authentication/Login';
import Signup from './authentication/Signup';
import { DynamicModule } from './DynamicModule';
import Footer from './footer/Footer';
import Header from './header/Header';
import { ReactComponent as LLogo } from './images/Lucy24_logo.svg';
import { appModules } from './settings/appModuleSettings';
import { shellApi } from './ShellApi';
import { ChangeThemeAction } from './themes/themeReducer/themeReducer';

export interface AppProps {
  changeTheme: (useDarkTheme: boolean) => ChangeThemeAction;
}

const App: FunctionComponent<AppProps> = ({ changeTheme }: AppProps) => {
  const { formatMessage } = useIntl();
  const loadingMessage: string = formatMessage({ id: 'fallback.loading' });
  useTheme();

  initializeIcons();

  const lucyIconStyle: CSSProperties = {
    width: '20px',
    height: '20px',
    overflow: 'visible',
  };

  registerIcons({
    icons: {
      Lucy: <LLogo style={lucyIconStyle} />,
    },
  });

  const appStyle: CSSProperties = {
    textAlign: 'center',
    height: '100vh',
  };

  const pageContainerStyle: CSSProperties = {
    position: 'relative',
    height: '100%',
  };

  // TODO: check for user product existance CLOUD-2148
  const userHasProduct = false;

  return (
    <div className="App" style={appStyle}>
      <BrowserRouter>
        <Switch>
          <Route exact path="/login">
            <Login />
          </Route>
          <Route exact path="/signup">
            <Signup />
          </Route>
          <Route path="/">
            <Stack
              grow
              styles={{
                root: {
                  height: '100%',
                  width: '100%',
                },
              }}
            >
              {(!location.pathname.includes('product') || userHasProduct) &&
                <Stack.Item>
                  <Header changeTheme={changeTheme} />
                </Stack.Item>
              }
              <Stack.Item shrink>
                <IdleMonitor />
              </Stack.Item>
              <Stack.Item
                verticalFill
                styles={{
                  root: {
                    height: '100%',
                    overflow: 'auto',
                  },
                }}
              >
                <main style={pageContainerStyle} role="main">
                  <ScrollablePane scrollbarVisibility={ScrollbarVisibility.auto}>
                    <Switch>
                      {appModules.map(appModule => (
                        <Route exact key={appModule.scope} path={appModule.routeName}>
                          <DynamicModule module={appModule} loadingMessage={loadingMessage} shellApi={shellApi} />
                        </Route>
                      ))}
                    </Switch>
                  </ScrollablePane>
                </main>
              </Stack.Item>
              {(!location.pathname.includes('product') || userHasProduct) &&
                <Stack.Item>
                  <Footer />
                </Stack.Item>
              }
            </Stack>
          </Route>
        </Switch>
      </BrowserRouter>
    </div>
  );
};

export default App;
