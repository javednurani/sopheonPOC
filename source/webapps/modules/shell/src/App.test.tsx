import { ThemeProvider } from '@fluentui/react-theme-provider';
import { darkTheme, lightTheme, messages } from '@sopheon/shared-ui';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { axe, toHaveNoViolations } from 'jest-axe';
import React from 'react';
import { IntlProvider } from 'react-intl';

import App, { AppProps } from './App';
import Footer from './footer/Footer';
import Header from './header/Header';

expect.extend(toHaveNoViolations);

describe('Testing the App component', () => {
  it('Render test for the App component', () => {
    const appProps: AppProps = {
      changeTheme: jest.fn(),
      setEnvironmentKey: jest.fn(),
    };

    const wrapper = mount(
      <IntlProvider locale="en" messages={messages.en}>
        <App {...appProps} />
      </IntlProvider>
    );
    expect(wrapper.find(Header)).toHaveLength(1);
    expect(wrapper.find(Header).props()).toStrictEqual(appProps);
    expect(wrapper.find('main')).toHaveLength(1);
    expect(wrapper.find('main').getDOMNode()).toHaveAttribute('role', 'main');
    expect(wrapper.find(Footer)).toHaveLength(1);
  });
  it('Accessibility test for the App component', async () => {
    const appProps: AppProps = {
      changeTheme: jest.fn(),
      setEnvironmentKey: jest.fn(),
    };

    const wrapper = mount(
      <IntlProvider locale="en" messages={messages.en}>
        <App {...appProps} />
      </IntlProvider>
    );
    //When the app is nested in the shell, we expect 0 axe violations
    const results = await axe(wrapper.getDOMNode());
    expect(results).toHaveNoViolations();
  });
});
