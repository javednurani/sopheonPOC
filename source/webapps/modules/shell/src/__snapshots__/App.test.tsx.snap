// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the App component App snapshot dark theme render test 1`] = `
<div
  className="root-160"
>
  <div
    className="App"
    style={
      Object {
        "height": "100vh",
        "textAlign": "center",
      }
    }
  >
    <div
      className="ms-Stack css-120"
    >
      <div
        className="ms-StackItem css-121"
      >
        <div>
          <header
            role="banner"
            style={
              Object {
                "borderBottom": "1px solid",
                "marginTop": "8px",
              }
            }
          >
            <div
              className="ms-Stack css-122"
            >
              <div
                className="ms-StackItem css-121"
              >
                <div
                  key=".0"
                  style={
                    Object {
                      "margin": "5px",
                    }
                  }
                  title="Sopheon"
                >
                  <ReactComponent
                    style={
                      Object {
                        "height": "100%",
                        "maxHeight": "200px",
                        "maxWidth": "200px",
                        "minHeight": "25px",
                        "minWidth": "25px",
                        "overflow": "visible",
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="ms-StackItem css-121"
              >
                <nav
                  role="navigation"
                >
                  <a
                    href="/product"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#61a39f",
                        "fontSize": "20px",
                      }
                    }
                  >
                    Home
                  </a>
                </nav>
              </div>
              <div
                className="ms-StackItem css-123"
              >
                <h1
                  className="page-title"
                  id="page-title"
                  key=".0"
                >
                  Template Title
                </h1>
              </div>
              <div
                className="ms-StackItem css-121"
              >
                <div
                  className="ms-Stack css-124"
                >
                  <div
                    className="ms-StackItem css-121"
                  >
                    Array [
                      "",
                      Array [
                        <button
                          className="ms-Button ms-Button--default root-162"
                          data-is-focusable={true}
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          onKeyPress={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseUp={[Function]}
                          type="button"
                        >
                          <span
                            className="ms-Button-flexContainer flexContainer-126"
                            data-automationid="splitbuttonprimary"
                          >
                            <span
                              className="ms-Button-textContainer textContainer-127"
                            >
                              <span
                                className="ms-Button-label label-129"
                                id="id__18"
                                key="id__18"
                              >
                                Log In
                              </span>
                            </span>
                          </span>
                        </button>,
                        "",
                      ],
                    ]
                  </div>
                  <div
                    className="ms-StackItem css-121"
                  >
                    <div
                      className="ms-Stack css-133"
                    >
                      Dark Theme
                      <div
                        className="ms-Toggle is-enabled root-134"
                      >
                        <div
                          className="ms-Toggle-innerContainer container-136"
                        >
                          <button
                            aria-checked={false}
                            aria-label="Dark Theme"
                            className="ms-Toggle-background pill-163"
                            data-is-focusable={true}
                            id="Toggle21"
                            onChange={[Function]}
                            onClick={[Function]}
                            role="switch"
                            type="button"
                          >
                            <span
                              className="ms-Toggle-thumb thumb-164"
                            />
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </header>
        </div>
      </div>
      <div
        className="ms-StackItem css-121"
      />
      <div
        className="ms-StackItem css-142"
      >
        <main
          key=".0"
          role="main"
          style={
            Object {
              "height": "100%",
              "position": "relative",
            }
          }
        >
          <div
            className="ms-ScrollablePane root-143"
          >
            <div
              className="stickyAbove-145"
              style={
                Object {
                  "height": 0,
                  "left": "0",
                  "right": "0px",
                  "top": "0",
                }
              }
            />
            <div
              className="ms-ScrollablePane--contentContainer contentContainer-144"
              data-is-scrollable={true}
            />
            <div
              className="stickyBelow-146"
              style={
                Object {
                  "bottom": "0px",
                  "height": 0,
                  "left": "0",
                  "right": "0px",
                }
              }
            >
              <div
                className="stickyBelowItems-147"
              />
            </div>
          </div>
        </main>
      </div>
      <div
        className="ms-StackItem css-121"
      >
        <div>
          <footer
            role="contentinfo"
            style={
              Object {
                "borderTop": "1px solid",
                "padding": "10px",
                "textAlign": "start",
              }
            }
          >
            <div
              className="ms-Stack css-122"
            >
              <div
                className="ms-StackItem css-148"
              >
                RSS Feeds | Sitemap | Privacy | Legal | Contact Us :: Copyright 2020-2021 Sopheon. All Rights Reserved
              </div>
              <div
                className="ms-StackItem css-149"
              >
                <span
                  key=".0"
                >
                  <i
                    aria-hidden="true"
                    className="root-151"
                    data-icon-name="LikeSolid"
                  >
                    
                  </i>
                  <i
                    aria-hidden="true"
                    className="root-153"
                    data-icon-name="CommentSolid"
                  >
                    
                  </i>
                  <i
                    aria-hidden="true"
                    className="root-155"
                    data-icon-name="LinkedInLogo"
                  >
                    
                  </i>
                  <i
                    aria-hidden="true"
                    className="root-157"
                    data-icon-name="TeamsLogo"
                  >
                    
                  </i>
                </span>
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing the App component App snapshot light theme render test 1`] = `
<div
  className="root-158"
>
  <div
    className="App"
    style={
      Object {
        "height": "100vh",
        "textAlign": "center",
      }
    }
  >
    <div
      className="ms-Stack css-120"
    >
      <div
        className="ms-StackItem css-121"
      >
        <div>
          <header
            role="banner"
            style={
              Object {
                "borderBottom": "1px solid",
                "marginTop": "8px",
              }
            }
          >
            <div
              className="ms-Stack css-122"
            >
              <div
                className="ms-StackItem css-121"
              >
                <div
                  key=".0"
                  style={
                    Object {
                      "margin": "5px",
                    }
                  }
                  title="Sopheon"
                >
                  <ReactComponent
                    style={
                      Object {
                        "height": "100%",
                        "maxHeight": "200px",
                        "maxWidth": "200px",
                        "minHeight": "25px",
                        "minWidth": "25px",
                        "overflow": "visible",
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="ms-StackItem css-121"
              >
                <nav
                  role="navigation"
                >
                  <a
                    href="/product"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#33596f",
                        "fontSize": "20px",
                      }
                    }
                  >
                    Home
                  </a>
                </nav>
              </div>
              <div
                className="ms-StackItem css-123"
              >
                <h1
                  className="page-title"
                  id="page-title"
                  key=".0"
                >
                  Template Title
                </h1>
              </div>
              <div
                className="ms-StackItem css-121"
              >
                <div
                  className="ms-Stack css-124"
                >
                  <div
                    className="ms-StackItem css-121"
                  >
                    Array [
                      "",
                      Array [
                        <button
                          className="ms-Button ms-Button--default root-125"
                          data-is-focusable={true}
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          onKeyPress={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseUp={[Function]}
                          type="button"
                        >
                          <span
                            className="ms-Button-flexContainer flexContainer-126"
                            data-automationid="splitbuttonprimary"
                          >
                            <span
                              className="ms-Button-textContainer textContainer-127"
                            >
                              <span
                                className="ms-Button-label label-129"
                                id="id__12"
                                key="id__12"
                              >
                                Log In
                              </span>
                            </span>
                          </span>
                        </button>,
                        "",
                      ],
                    ]
                  </div>
                  <div
                    className="ms-StackItem css-121"
                  >
                    <div
                      className="ms-Stack css-133"
                    >
                      Dark Theme
                      <div
                        className="ms-Toggle is-enabled root-134"
                      >
                        <div
                          className="ms-Toggle-innerContainer container-136"
                        >
                          <button
                            aria-checked={false}
                            aria-label="Dark Theme"
                            className="ms-Toggle-background pill-137"
                            data-is-focusable={true}
                            id="Toggle15"
                            onChange={[Function]}
                            onClick={[Function]}
                            role="switch"
                            type="button"
                          >
                            <span
                              className="ms-Toggle-thumb thumb-138"
                            />
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </header>
        </div>
      </div>
      <div
        className="ms-StackItem css-121"
      />
      <div
        className="ms-StackItem css-142"
      >
        <main
          key=".0"
          role="main"
          style={
            Object {
              "height": "100%",
              "position": "relative",
            }
          }
        >
          <div
            className="ms-ScrollablePane root-143"
          >
            <div
              className="stickyAbove-145"
              style={
                Object {
                  "height": 0,
                  "left": "0",
                  "right": "0px",
                  "top": "0",
                }
              }
            />
            <div
              className="ms-ScrollablePane--contentContainer contentContainer-144"
              data-is-scrollable={true}
            />
            <div
              className="stickyBelow-146"
              style={
                Object {
                  "bottom": "0px",
                  "height": 0,
                  "left": "0",
                  "right": "0px",
                }
              }
            >
              <div
                className="stickyBelowItems-147"
              />
            </div>
          </div>
        </main>
      </div>
      <div
        className="ms-StackItem css-121"
      >
        <div>
          <footer
            role="contentinfo"
            style={
              Object {
                "borderTop": "1px solid",
                "padding": "10px",
                "textAlign": "start",
              }
            }
          >
            <div
              className="ms-Stack css-122"
            >
              <div
                className="ms-StackItem css-148"
              >
                RSS Feeds | Sitemap | Privacy | Legal | Contact Us :: Copyright 2020-2021 Sopheon. All Rights Reserved
              </div>
              <div
                className="ms-StackItem css-149"
              >
                <span
                  key=".0"
                >
                  <i
                    aria-hidden="true"
                    className="root-151"
                    data-icon-name="LikeSolid"
                  >
                    
                  </i>
                  <i
                    aria-hidden="true"
                    className="root-153"
                    data-icon-name="CommentSolid"
                  >
                    
                  </i>
                  <i
                    aria-hidden="true"
                    className="root-155"
                    data-icon-name="LinkedInLogo"
                  >
                    
                  </i>
                  <i
                    aria-hidden="true"
                    className="root-157"
                    data-icon-name="TeamsLogo"
                  >
                    
                  </i>
                </span>
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
  </div>
</div>
`;
